crypto/cmp/cmp_msg.c

```diff
@@ -817,8 +817,13 @@ OSSL_CMP_MSG *ossl_cmp_certConf_new(OSSL_CMP_CTX *ctx, int fail_info,
     if ((certStatus = OSSL_CMP_CERTSTATUS_new()) == NULL)
         goto err;
     /* consume certStatus into msg right away so it gets deallocated with msg */
-    if (!sk_OSSL_CMP_CERTSTATUS_push(msg->body->value.certConf, certStatus))
+    if (sk_OSSL_CMP_CERTSTATUS_push(msg->body->value.certConf, certStatus) < 1) {
```

Comment:
@DDvO What frees this stack in the normal case?

Reply:
As usual with ASN.1 structures, on success ownership of field contents is passed to the containing structure, so the (stack of) `OSSL_CMP_CERTSTATUS` will be freed along with the `msg` containing it.
This is is even made (rather) explicit here by the comment above.